src/xdp-utils.c

```diff
@@ -2349,3 +2349,67 @@ xdp_validate_serialized_icon (GVariant  *v,
 
   return TRUE;
 }
+
+#if G_ENCODE_VERSION (GLIB_MAJOR_VERSION, GLIB_MINOR_VERSION) >= G_ENCODE_VERSION (2, 66)
+gchar *
+xdp_transform_remote_uri_into_local (const gchar *uri)
+{
+  gchar *pathtext = NULL;
+  gchar *uristring = NULL;
+  gchar *fuse_mountpoint = NULL;
+  GUri *file_uri = NULL;
+  GString *gvfs_folder = NULL;
+
+  file_uri = g_uri_parse (uri, G_URI_FLAGS_NONE, NULL);
+  if (file_uri == NULL)
+    {
+      return g_strdup(uri);
+    }
+  if ((0 != g_strcmp0 (g_uri_get_scheme (file_uri), "file")) &&
+      (g_uri_get_host (file_uri) != NULL))
```

Comment:
This would be easier to read if you returned on all the cases where this function doesn't work, and the contents of the function don't need to be indented then. i.e.

```
if (file_uri == NULL || g_strcmp0 (g_uri_get_scheme (file_uri), "file") == 0 || g_uri_get_host (file_uri) == NULL)
  {
    return g_strdup (uri);
  }

// Do all the remote URI code processing here
```

Reply:
Done.